clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -556,10 +557,15 @@ public static Set<TopicPartition> generateRandomTopicPartitions(int numTopic, in
      * @return The caught exception cause
      */
     public static <T extends Throwable> T assertFutureThrows(Future<?> future, Class<T> exceptionCauseClass) {
-        ExecutionException exception = assertThrows(ExecutionException.class, future::get);
-        assertInstanceOf(exceptionCauseClass, exception.getCause(),
-            "Unexpected exception cause " + exception.getCause());
-        return exceptionCauseClass.cast(exception.getCause());
+        try {
+            future.get(5, TimeUnit.SECONDS);
+            ExecutionException exception = assertThrows(ExecutionException.class, future::get);
+            assertInstanceOf(exceptionCauseClass, exception.getCause(),
+                    "Unexpected exception cause " + exception.getCause());
+            return exceptionCauseClass.cast(exception.getCause());
+        } catch (Exception ignored) {
+            return null;
+        }
```

Comment:
Before this change, when future.get throws "non-ExecutionException", it'll fail. But after the change, it'll pass. This changes the original semantics. I think we can use `catch` to assert what we expected. Ex:

```
try {
  future.get(5, sec);
  fail("expected to throw ExecutionException...");
} catch (TimeoutException e) {
  fail("timeout waiting....");
} catch (ExecutionException e) {
  ...
} catch ...


```

WDYT? 
```

Reply:
AI saw there's another `future.get` without timeout in L583. Please also fix it. Thanks.

Reply:
Thanks for the comments, would apply.